API/src/main/java/de/brandwatch/minianalytics/api/controller/QueryController.java

```diff
@@ -26,31 +29,31 @@ public QueryController(QueryService queryService) {
     }
 
     @PostMapping(value = "/queries")
-    public ResponseEntity query(@RequestParam String query) {
+    public Query query(@RequestParam String query) {
         try {
             logger.info("POST /queries: {}", query);
-            return ResponseEntity.status(200).body(gson.toJson(queryService.createQuery(query)));
+            return queryService.createQuery(query);
         } catch (Exception e) {
             throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
         }
     }
 
     @GetMapping(value = "/queries")
-    public ResponseEntity queries() {
+    public List<Query> queries() {
         try {
             logger.info("GET /queries");
-            return ResponseEntity.status(200).body(gson.toJson(queryService.getAllQueries()));
+            return queryService.getAllQueries();
         } catch (Exception e) {
             throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
         }
     }
 
     @GetMapping(value = "/queries/{queryID}")
-    public ResponseEntity singleQuery(@PathVariable("queryID") String queryID) {
+    public Optional<Query> singleQuery(@PathVariable("queryID") String queryID) {
```

Comment:
How does this response look like? And how does it look like if the optional is empty?

I would suggest something like this instead:

```java
@GetMapping(value = "/queries/{queryID}")
public Query singleQuery(@PathVariable("queryID") String queryID) {
    try {
        logger.info("GET /queries/" + queryID);
        return queryService
                .getQueryByID(queryID)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NO_CONTENT, "<some message>"));
    } catch (Exception e) {
        throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "", e);
    }
}
```